packages/react-charting/UnitTests/HeatMapChartUT.test.tsx

```diff
@@ -0,0 +1,676 @@
+    const result = instance._getStringFormattedNumber('10000', '.2~s');
+    expect(result).toEqual('10k');
+  });
+
+  test('Should return proper string formatted number for empty data', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('');
+    expect(result).toEqual('0');
+  });
+
+  test('Should return proper string formatted number for string value', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedNumber('Test1');
+    expect(result).toEqual('NaN');
+  });
+});
+
+runTest('_getStringFormattedDate', () => {
+  test('Should return proper string formatted date for date point', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedDate('100000000');
+    expect(result).toEqual('Jan/02');
+  });
+
+  test('Should return proper string formatted date for empty point', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedDate('');
+    expect(result).toEqual('Jan/01');
+  });
+
+  test('Should return proper string formatted date for date point', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getStringFormattedDate('100000000', '%b/%d/%Y');
+    expect(result).toEqual('Jan/02/1970');
+  });
+});
+
+runTest('_getXAxisDataPoints', () => {
+  test('Should return proper xAxis data points for string points', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    const result = instance._getXAxisDataPoints({ B: '1', A: '1' });
+    expect(result[0]).toEqual('A');
+    expect(result[1]).toEqual('B');
+  });
+
+  test('Should return proper xAxis data points for numeric points', () => {
+    const instance = new HeatMapChartBase({
+      data: emptyData,
+      domainValuesForColorScale: [],
+      rangeValuesForColorScale: [],
+    });
+    expect(instance).toBeDefined();
+    instance._xAxisType = XAxisTypes.NumericAxis;
+    const result = instance._getXAxisDataPoints({ 100: '1', 30: '1' });
+    expect(result[0]).toEqual('30');
```

Comment:
shoudln't result[0] equal 100 and result[1] equal 30

Reply:
returning data in sorted format.